allensdk/api/cloud_cache/cloud_cache.py

```diff
@@ -1107,3 +1220,82 @@ def _download_manifest(self, manifest_name: str):
 
     def _download_file(self, file_attributes: CacheFileAttributes) -> bool:
         raise NotImplementedError()
+
+
+class StaticLocalCache(BasicLocalCache):
+    """A class to handle accessing data that has already been downloaded
+    locally and whose directory structure and/or contained files are not
+    expected to be changed in any way. Does NOT support multiple manifest
+    versions for a given dataset.
+
+    Example intended use case:
+        Calling
+        VisualBehaviorOphysProjectCache.from_local_cache(use_static_cache=True)
+        where the cache directory is a mounted S3 public bucket.
+
+    Parameters
+    ----------
+    cache_dir: str or pathlib.Path
+        Path to the directory where data will be stored on the local system
+
+    project_name: str
+        the name of the project this cache is supposed to access. This will
+        be the root directory for all files stored in the bucket.
+
+    ui_class_name: Optional[str]
+        Name of the class users are actually using to maniuplate this
+        functionality (used to populate helpful error messages)
+    """
+
+    def __init__(self, cache_dir, project_name, ui_class_name=None):
+        super().__init__(cache_dir=cache_dir, project_name=project_name,
+                         ui_class_name=ui_class_name)
+
+    def _list_all_manifests(self) -> list:
```

Comment:
Because the S3 bucket contains all of the versions of the manifest, this will return all versions of the manifest. Given that users don't actually have the ability to load any but the latest manifest, should this version only return `self.latest_manifest_file`?

Reply:
@danielsf This is a case of the `_list_all_manifests()` method name probably being too specific. Unfortunately, I was pretty hesitant to change it since both `BasicLocalCache` and `CloudCacheBase` call `_list_all_manifests()` and not some other method name. If you look at the implementation (line 1267) it in only returns the latest manifest.